### PR TITLE
EN-2313 Improvements to help with Ratings from DE plugin in components

### PR DESCRIPTION
**Pull request by rzsouza**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by rzsouza** [comment]
restest this please

**Comment by entando-service** [comment]
All is well

File: src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java [hunk 1]
```diff
@@ -190,6 +198,38 @@ public void delete(long id) {
         this.executeQueryWithoutResultset(DELETE_RATING, id);
     }
 
+    public Optional<DERating> findByReviewerAndComponent(String reviewerId, String componentId) {
```
Comment (zonia3000) [inline src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java]: Add `@Override` annotation

File: src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import org.springframework.stereotype.Repository;
 
 import javax.sql.DataSource;
+import java.security.*;
```
Comment (zonia3000) [inline src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java]: Unused import

File: src/main/java/org/entando/entando/web/digitalexchange/component/DEComponentExtensionController.java [hunk 3]
```diff
@@ -8,14 +8,16 @@
...
+import org.entando.entando.web.digitalexchange.rating.DEComponentRatingRequest;
 import org.springframework.core.io.InputStreamResource;
 import org.springframework.http.*;
 import org.springframework.validation.BindingResult;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.multipart.MultipartFile;
 
 import javax.validation.Valid;
-import java.util.Optional;
+import java.security.*;
```
Comment (zonia3000) [inline src/main/java/org/entando/entando/web/digitalexchange/component/DEComponentExtensionController.java]: Unused import

**Comment by entando-service** [comment]
All is well

File: src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java (hunk 2 above)
Comment (rzsouza) [reply src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java]: Fixed

File: src/main/java/org/entando/entando/web/digitalexchange/component/DEComponentExtensionController.java (hunk 3 above)
Comment (rzsouza) [reply src/main/java/org/entando/entando/web/digitalexchange/component/DEComponentExtensionController.java]: Fixed

File: src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java (hunk 1 above)
Comment (rzsouza) [reply src/main/java/org/entando/entando/aps/system/services/digitalexchange/rating/DERatingsDaoImpl.java]: Fixed

**Comment by entando-service** [comment]
All is well

**Pull request merged**